wrappers/rust/oneds-telemetry/src/types.rs

```diff
@@ -0,0 +1,182 @@
+use std::{
+    collections::HashMap,
+    ffi::{CStr, CString},
+    os::raw::c_char,
+};
+
+use chrono::{DateTime, Utc};
+use log::{debug, error};
+use once_cell::sync::Lazy;
+
+use cpp_client_telemetry_sys::{evt_prop, evt_prop_v, evt_prop_t_TYPE_STRING, evt_prop_t_TYPE_TIME, evt_prop_t_TYPE_INT64};
+
+#[derive(Clone, Debug)]
+pub struct StringProperty {
+    bytes: Vec<u8>,
+}
+
+impl StringProperty {
+    pub fn new(value: &str) -> Self {
```

Comment:
This function appears to allocate three copies of the string.  Can this be reduced to one allocation?

```
pub fn new(value: &str) -> Self {
  let mut bytes = Vec::with_capacity(value.len() + 1);
  bytes.extend_from_slice(value);
  // last element is implicitly zero
  Self { bytes }
}
```